disk-utils/blockdev.c

```diff
@@ -254,12 +254,13 @@ static int find_cmd(char *s)
 
 static void do_commands(int fd, char **argv, int d);
 static void report_header(void);
-static void report_device(char *device, int quiet);
-static void report_all_devices(void);
+static void report_device(char *device, int quiet, int *error);
+static void report_all_devices(int *error);
 
 int main(int argc, char **argv)
 {
 	int fd, d, j, k;
+	int report_error = 0;
```

Comment:
It does not have to be defined for whole main(), you can move it to `if (!strcmp(argv[1], "--report")) ` block.